driver/src/main/java/org/neo4j/driver/internal/RoutingDriver.java

```diff
@@ -243,112 +59,56 @@ public Session session()
-                new RoutingErrorHandler()
-                {
-                    @Override
-                    public void onConnectionFailure( BoltServerAddress address )
-                    {
-                        forget( address );
-                    }
-
-                    @Override
-                    public void onWriteFailure( BoltServerAddress address )
-                    {
-                        clusterView.removeWriter( address );
-                    }
-                } );
+        return new RoutingNetworkSession( new NetworkSession( connection ), mode, connection.address(), loadBalancer );
     }
 
     private Connection acquireConnection( AccessMode role )
     {
-        //Potentially rediscover servers if we are not happy with our current knowledge
-        checkServers();
-
         switch ( role )
         {
         case READ:
-            return acquireReadConnection();
+            return loadBalancer.acquireReadConnection();
         case WRITE:
-            return acquireWriteConnection();
+            return loadBalancer.acquireWriteConnection();
         default:
             throw new ClientException( role + " is not supported for creating new sessions" );
         }
     }
 
-    private Connection acquireReadConnection()
-    {
-        int numberOfServers = clusterView.numberOfReaders();
-        for ( int i = 0; i < numberOfServers; i++ )
-        {
-            BoltServerAddress address = clusterView.nextReader();
-            try
-            {
-                return connections.acquire( address );
-            }
-            catch ( ServiceUnavailableException e )
-            {
-                forget( address );
-            }
-        }
-
-        throw new SessionExpiredException( "Failed to connect to any read server" );
-    }
-
-    private Connection acquireWriteConnection()
-    {
-        int numberOfServers = clusterView.numberOfWriters();
-        for ( int i = 0; i < numberOfServers; i++ )
-        {
-            BoltServerAddress address = clusterView.nextWriter();
-            try
-            {
-                return connections.acquire( address );
-            }
-            catch ( ServiceUnavailableException e )
-            {
-                forget( address );
-            }
-        }
-
-        throw new SessionExpiredException( "Failed to connect to any write server" );
-    }
-
     @Override
     public void close()
     {
         try
         {
-            connections.close();
+            loadBalancer.close();
         }
         catch ( Exception ex )
         {
             log.error( format( "~~ [ERROR] %s", ex.getMessage() ), ex );
         }
     }
 
-    //For testing
-    public Set<BoltServerAddress> routingServers()
+    Set<BoltServerAddress> routingServers()
     {
-        return clusterView.routingServers();
+        // TODO: the tests that use this should be testing for effect instead
```

Comment:
It is ignored just because of current TC limitations. We still want all those tests to run locally, Zhen enabled them just the other day but had to revert because of boltkit issues. The plan is still that those tests should be unignored asap.